tuf/pyca_crypto_keys.py

```diff
@@ -0,0 +1,1017 @@
+      A string that indicates the signature algorithm used to generate
+      'signature'.  'RSASSA-PSS' is currently supported.
+
+    public_key:
+      The RSA public key, a string in PEM format.
+
+    data:
+      Data used by tuf.keys.create_signature() to generate
+      'signature'.  'data' (a string) is needed here to verify 'signature'.
+
+  <Exceptions>
+    tuf.UnknownMethodError.  Raised if the signing method used by
+    'signature' is not one supported by tuf.keys.create_signature().
+    
+    tuf.FormatError. Raised if 'signature', 'signature_method', or 'public_key'
+    are improperly formatted.
+
+  <Side Effects>
+    pyca/cryptography's RSAPublicKey.verifier() called to do the actual
+    verification.
+
+   <Returns>
+    Boolean.  True if the signature is valid, False otherwise.
+  """
+  
+  # Does 'public_key' have the correct format?
+  # This check will ensure 'public_key' conforms to 'tuf.formats.PEMRSA_SCHEMA'.
+  # Raise 'tuf.FormatError' if the check fails.
+  tuf.formats.PEMRSA_SCHEMA.check_match(public_key)
+
+  # Does 'signature_method' have the correct format?
+  tuf.formats.NAME_SCHEMA.check_match(signature_method)
+
+  # Does 'signature' have the correct format?
+  tuf.formats.PYCRYPTOSIGNATURE_SCHEMA.check_match(signature)
+  
+  # What about 'data'?
+  tuf.formats.DATA_SCHEMA.check_match(data)
+
+  # Verify whether the private key of 'public_key' produced 'signature'.
+  # Before returning the 'valid_signature' Boolean result, ensure 'RSASSA-PSS'
+  # was used as the signing method.
+  valid_signature = False
+
+  # Verify the signature with pyca/cryptography if the signature method is
+  # valid, otherwise raise 'tuf.UnknownMethodError'.
+  if signature_method == 'RSASSA-PSS':
+    try:
+      public_key_object = serialization.load_pem_public_key(public_key.encode('utf-8'),
+                                                     backend=default_backend())
+      
+      # 'salt_length' is set to the digest size of the hashing algorithm (to
+      # match the default size used by 'tuf.pycrypto_keys.py').
+      verifier = public_key_object.verifier(signature,
+                                  padding.PSS(mgf=padding.MGF1(hashes.SHA256()),
+                                  salt_length=hashes.SHA256().digest_size), 
+                                  hashes.SHA256())
+
+      verifier.update(data.encode('utf-8'))
+      
+      # verify() raises 'cryptograpahy.exceptions.InvalidSignature' if the
+      # signature is invalid.
+      try: 
+        verifier.verify()
+        valid_signature = True 
+      
+      except cryptography.exceptions.InvalidSignature as e:
+        pass
+ 
+    # Raised by load_pem_public_key(). 
+    except ValueError:
+      raise tuf.CryptoError('The PEM could not be decoded successfully.')
+
+    # Raised by load_pem_public_key().
+    except cryptography.exceptions.UnsupportedAlgorithm:
+      raise tuf.Cryptography('The private key type is not supported.')
```

Comment:
`tuf.Cryptography` doesn't seem to be an existing exception class. Probably meant `tuf.CryptoError`.